src/components/element-editor/validation/helper.tsx

```diff
@@ -9,6 +9,21 @@ export const getValidationOptionsByType = (
     validation: IValidation[],
     elementType: EElementType
 ) => {
+    const getFilteredValidationTypes = (
+        validation: IValidation[],
+        validationTypes: string[],
+        ignoredTypes?: string[]
+    ): string[] => {
+        const hasType = (type: string) =>
+            validation.some((val) => val.validationType === type);
+
+        const filteredValidationTypes = validationTypes.filter(
+            (type) => !hasType(type) || ignoredTypes?.includes(type)
+        );
+
+        return filteredValidationTypes;
+    };
+
```

Comment:
- move this function out of `getValidationOptionsByType`
- change `validationTypes` and `ignoredTypes` typings to `EValidationType[]`